### PR TITLE
Use autossh instead of ssh

### PR DESCRIPTION
**Pull request by bubak4**
For extra reliability I can recommend using autossh instead of ssh (minor change in Dockerfile).

I have also added a simple (GNU) Makefile for container build and updated doc accordingly.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,15 +9,17 @@
 #
 ###
 
-FROM alpine:3.2
+FROM alpine:latest
```
Comment (fruitl00p) [inline Dockerfile:12]: I can agree on using a later alpine version, but `:latest` is not recommended. Could you change this to 3.5? That at least reflects the actual used version ;)

File: Dockerfile [hunk 2]
```diff
@@ -9,15 +9,17 @@
...
 MAINTAINER Kingsquare <docker@kingsquare.nl>
+MAINTAINER bubak4 <mar@centrum.cz>
```
Comment (fruitl00p) [inline Dockerfile:14]: We might want to remove the maintainers as a whole .. see docker/docker#25466
Possibly include a label referring to this repo ... that has more info anyway on the subject (code authors etc)

File: Dockerfile [hunk 3]
```diff
@@ -9,15 +9,17 @@
...
-RUN apk add --update openssh-client && rm -rf /var/cache/apk/*
+# Install the autossh
+RUN apk add --update autossh && rm -rf /var/cache/apk/*
 
 VOLUME ["/ssh-agent"]
-EXPOSE 2222 
-ENTRYPOINT ["/usr/bin/ssh", "-T", "-N", "-o", "StrictHostKeyChecking=false", "-o", "ServerAliveInterval=180", "-L"]
+# for ambassador mode
+EXPOSE 2222
+ENTRYPOINT ["/usr/bin/autossh", "-M", "0", "-T", "-N", "-oStrictHostKeyChecking=no", "-L"]
```
Comment (fruitl00p) [inline Dockerfile:25]: if replacing ssh with autossh, you still want to keep the `ServerAliveInterval=180` setting... (as you've disabled the monitoring rightly so)

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Time-stamp: < Makefile (2017-01-10 10:03) >
+BUILD=build
+NAME=docker-tunnel
+TAG=bubak4
```
Comment (fruitl00p) [inline Makefile:4]: i understand you're tag, but this shouldnt be in the original repo IMHO

File: README.md [hunk 5]
```diff
@@ -31,35 +31,48 @@ This method allows for using this image as an ambassador to other (secure) serve
...
+   - Use `autossh` instead of simple `ssh` for extra stability of the tunnel
+   - Provided sample `Makefile` to automate the build process
+
+	```SSH_CMD="*:6379:localhost:6379 martin@172.17.0.1" make build-container```
+
+   - As original, the assumption is, that local `ssh-agent` holds the
+     required identity files.  Better solution probably would be to
+     generate new ssh key (`ssh-keygen`) and use the ssh `-i` option to
+     provide the identity directly
+
```
Comment (fruitl00p) [inline README.md:59]: i understand the recommendation, but i dont see how thats better? Its up to the local ssh-agent in the original case to have the correct identities... (and using the -i option would require volume-ing that in as well... )

**Review by fruitl00p — CHANGES_REQUESTED** [review]
I've looked over your proposal... Please review these and see if we can merge at least the core: autossh + updated alpine...

File: Dockerfile (hunk 1 above)
Comment (bubak4) [reply Dockerfile:12]: no problem, will change to 3.5

File: Dockerfile (hunk 2 above)
Comment (bubak4) [reply Dockerfile:14]: no problem, I just copied the line with my name as I thought you do it that way, I do not use it either, will remove

File: Dockerfile (hunk 3 above)
Comment (bubak4) [reply Dockerfile:25]: I use autossh reliably without the ServerAliveInterval, but I did some search on the Internet and it is recommended to be used, so be it :-)

File: README.md (hunk 5 above)
Comment (bubak4) [reply README.md:59]: In my deployments I am just used to import project specific keys -- I like to be explicit what keys are used (and make them part of the container) than to rely on local agent to provide the identities.   But I have no intention to change that in your repository, as I have not yet found good enough solution to generate and provide those keys -- it always requires a bit of shell scripting

File: Makefile (hunk 4 above)
Comment (bubak4) [reply Makefile:4]: well, in fact I did not expect you will integrate my Makefile at all, so I was not extra careful -- it is a bit Linux specific as this is my platform of choice -- I just included the Makefile just for my own reference as I use Makefile a lot to automate docker container builds.  I will make it a bit nicer, though it always involves a bit of work to modify it specific needs.

Feel free to ignore the Makefile.

**Comment by bubak4** [comment]
Sorry it took longer than I expected, I was quite busy those days.  Closing this request & creating another one with your suggestions incorporated.

**Pull request closed without merging**